src/core/utils.js

```diff
@@ -1,16 +1,25 @@
 'use strict';
 
-function getMantissaDecimalString(bignum) {
-  let mantissa = bignum.toPrecision(16)
-    .replace(/\./, '')      // remove decimal point
+// returns the mantissa from the passed in string,
+// adding zeros until it has 16 sd
+function getMantissa16FromString(decimalString) {
+  let mantissa = decimalString.replace(/\./, '')      // remove decimal point
     .replace(/e.*/, '')     // remove scientific notation
     .replace(/^0*/, '');    // remove leading zeroes
+  if (mantissa.length > 16) {
+    const rounded = Number(mantissa).toFixed(16);
```

Comment:
Well truncate is equivalent to round down. You could argue that we should use the same rounding mode as the rest of the program, which is round half up. But because this function is basically getting the first 16 digits of the mantissa, round down seems more natural to me. It could go either way though.